packages/utils/src/optionsList.ts

```diff
@@ -13,8 +12,8 @@ export default function optionsList(
   schema: RJSFSchema
 ): EnumOptionsType[] | undefined {
   if (schema.enum) {
-    return schema.enum.map((value, i) => {
-      const label = (schema.enumNames && schema.enumNames[i]) || String(value);
```

Comment:
Consider restoring the old `hack` that I removed since I'd like to not restore the change to the type. Also add a console.warn() about `enumNames` being deprecated, to be removed in the next major release